### PR TITLE
Removing regex expressions that crash Unity

### LINKED ISSUE #334: Exceptions on 2017.1 in .Net 4.6
```
ArgumentException: parsing "^(?<active>\*)?\s+(?<name>[\w\d\/\-_]+)\s*(?:[a-z|0-9]{7} \[(?<tracking>[\w\d\/\-\_]+)\])?" - Unrecognized escape sequence \_.
System.Text.RegularExpressions.RegexParser.ScanCharEscape () (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
System.Text.RegularExpressions.RegexParser.ScanCharClass (System.Boolean caseInsensitive, System.Boolean scanOnly) (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
System.Text.RegularExpressions.RegexParser.CountCaptures () (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
System.Text.RegularExpressions.RegexParser.Parse (System.String re, System.Text.RegularExpressions.RegexOptions op) (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
System.Text.RegularExpressions.Regex..ctor (System.String pattern, System.Text.RegularExpressions.RegexOptions options, System.TimeSpan matchTimeout, System.Boolean useCache) (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
System.Text.RegularExpressions.Regex..ctor (System.String pattern) (at <f044fe2c9e7e4b8e91984b41f0cf0b04>:0)
GitHub.Unity.Utility..cctor () (at C:/Users/Spade/Projects/GitHub/Unity_master/src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Utility.cs:17)
Rethrow as TypeInitializationException: The type initializer for 'GitHub.Unity.Utility' threw an exception.
GitHub.Unity.Styles.get_SmallLogo () (at C:/Users/Spade/Projects/GitHub/Unity_master/src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Styles.cs:681)
GitHub.Unity.Window.OnEnable () (at C:/Users/Spade/Projects/GitHub/Unity_master/src/UnityExtension/Assets/Editor/GitHub.Unity/UI/Window.cs:95)
UnityEditorInternal.InternalEditorUtility:LoadSerializedFileAndForget(String)
UnityEditor.WindowLayout:LoadWindowLayout(String, Boolean)
```

### PR DESCRIPTION
**Pull request by shana**
Fixes #334

From #335:
> Initially for #145 in #310 I went to remove these regex expressions as they are not used.
> Testing that, it worked.
> I went back and tried to fix the stored procedures.
> Resharper believed \_ to be incorrect. Expresso (a .net regex tool) didn't seem to mind it.
> I assumed that was the error and changed it to _. After that I failed to test appropriately.

### CODE REVIEW

**Comment by StanleyGoldman** [comment]
I'll create an example repository and a bug for people at Unity to diagnose..

**Comment by StanleyGoldman** [comment]
A lot of the regular expressions and functions in that class are not being used...
Do we want to clean that up?

**Comment by shana** [comment]
Yeah, we should probably throw away all that code, we don't need it anymore.

File: src/UnityExtension/Assets/Editor/GitHub.Unity/GitHub.Unity.csproj [hunk 1]
```diff
@@ -91,8 +91,6 @@
     <Compile Include="Misc\Utility.cs" />
     <Compile Include="Services\AuthenticationService.cs" />
     <Compile Include="Services\StatusService.cs" />
-    <Compile Include="Tasks\EvaluateProjectConfigurationTask.cs" />
-    <Compile Include="Tasks\TaskException.cs" />
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/GitHub.Unity.csproj]: A single comment

Comment (shana) [reply src/UnityExtension/Assets/Editor/GitHub.Unity/GitHub.Unity.csproj]: A review comment

File: src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Utility.cs [hunk 2]
```diff
@@ -1,75 +1,14 @@
...
 using System.Reflection;
-using System.Text.RegularExpressions;
 using UnityEditor;
 using UnityEngine;
 
 namespace GitHub.Unity
 {
     class Utility : ScriptableObject
     {
-        private static readonly ILogging logger = Logging.GetLogger<Utility>();
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Utility.cs]: Another review comment

File: src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Utility.cs [hunk 3]
```diff
@@ -1,75 +1,14 @@
...
-
-        public const string StatusRenameDivider = "->";
-        public static readonly Regex ListBranchesRegex =
-            new Regex(@"^(?<active>\*)?\s+(?<name>[\w\d\/\-_]+)\s*(?:[a-z|0-9]{7} \[(?<tracking>[\w\d\/\-\_]+)\])?");
-        public static readonly Regex ListRemotesRegex =
-            new Regex(@"(?<name>[\w\d\-_]+)\s+(?<url>https?:\/\/(?<login>(?<user>[\w\d]+)(?::(?<token>[\w\d]+))?)@(?<host>[\w\d\.\/\%]+))\s+\((?<function>fetch|push)\)");
-        public static readonly Regex LogCommitRegex = new Regex(@"commit\s(\S+)");
-        public static readonly Regex LogMergeRegex = new Regex(@"Merge:\s+(\S+)\s+(\S+)");
```
Comment (shana) [inline src/UnityExtension/Assets/Editor/GitHub.Unity/Misc/Utility.cs]: Another comment

**Comment by StanleyGoldman** [comment]
I deleted some test comments @shana

**Pull request merged**